### ISSUE TITLE
containerd integration: implement automatic snapshotted selection

### ISSUE THREAD

**Issue by thaJeztah** [moby/moby#44076]
### Description

With Graphdrivers, the docker daemon automatically selects the most suitable (opinionated / configurable) storage driver to use by 

- checking what the previously used storage-driver was
- trying each storage-driver, and if supported (based on backing filesystem, distro, etc) picks the first one
- falling back to `vfs` as last resort

With the containerd integration we should have a similar selection to make things "work" out of the box. Currently, we use `containerd.DefaultSnapshotter` as a default, but this const is hard-coded;

- Linux: [`"overlayfs"`](https://github.com/containerd/containerd/blob/f0a32c66dad1e9de716c9960af806105d691cd78/snapshotter_default_linux.go#L23)
- Windows [`"windows"`](https://github.com/containerd/containerd/blob/f0a32c66dad1e9de716c9960af806105d691cd78/snapshotter_default_windows.go#L23)
- Unix (freeBSD, openBSD, macOS) [`"native"`](https://github.com/containerd/containerd/blob/6fa958853197af31ffd1225a6c300589e5068667/snapshotter_default_unix.go#L26)

We should:

- Get a list of snapshotters supported by containerd (containerd checks if they can be loaded)
- Pick the first one from the list, in an (opinionated) order.
- Consider if we want (e.g.) `aufs` and/or `devicemapper` to be part of _automatic_ selection, or only "manual"

**Comment by AkihiroSuda** [comment]
> Consider if we want (e.g.) aufs and/or devicemapper to be part of automatic selection, or only "manual"

aufs is deprecated and shouldn't be part of automatic selection

**Comment by thaJeztah** [comment]
> > Consider if we want (e.g.) aufs and/or devicemapper to be part of automatic selection, or only "manual"
> 
> aufs is deprecated and shouldn't be part of automatic selection

Yes, I didn't want it in the  default list, but didn't realise it was deprecated in containerd as well. 
I noticed it was still loaded by default, and still requires `no_aufs` to opt-out, so thought it wasn't, [but it's deprecated indeed](https://github.com/containerd/containerd/blob/f033f6ff85ce397202ee69de22b4e3a4bf4093f5/docs/snapshotters/README.md#L19);

> aufs: AUFS. Deprecated since containerd 1.5. Planned to be removed in containerd 2.0. See also https://github.com/containerd/aufs .

**Comment by 26tanishabanik** [comment]
@thaJeztah , can I work on this issue, if you can guide me where to start with?
